integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHorizontalPodAutoscalerCustomMetrics.java

```diff
@@ -314,19 +322,41 @@ void testHPAWithCustomMetrics() {
 
     try {
       checkPodDeleted(managedServerPrefix + 3, domainUid, domainNamespace);
-    } catch (Exception ex) {
+    } catch (Throwable ex) {
```

Comment:
checkPodDeleted() does not throw exception. You will not hit the catch block after the timeout. 

You can rewrite the checkPodDeleted() method and follow the similar logic in testUntil() method below to add the check after the timeout.

 public static void testUntil(ConditionFactory conditionFactory, Callable<Boolean> conditionEvaluator,
                               LoggingFacade logger, String msg, Object... params) {
    try {
      conditionFactory
          .conditionEvaluationListener(createConditionEvaluationListener(logger, msg, params))
          .until(conditionEvaluator);
    } catch (ConditionTimeoutException timeout) {
      throw new TimeoutException(MessageFormat.format("Timed out waiting for: " + msg, params), timeout);
    }
  }

Reply:
rewrited